core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsCredentialsProviderChain.java

```diff
@@ -84,21 +86,33 @@ public static AwsCredentialsProviderChain of(AwsCredentialsProvider... awsCreden
         return builder().credentialsProviders(awsCredentialsProviders).build();
     }
 
+    /**
+     * Create an AWS credentials provider chain with default configuration that checks the given credential providers.
+     * @param awsCredentialsProviders The credentials providers that should be checked for credentials, in the order they should
+     *                                be checked.
+     * @return A credential provider chain that checks the provided credential providers in order.
+     */
+    public static AwsCredentialsProviderChain of(IdentityProvider<? extends AwsCredentialsIdentity>... awsCredentialsProviders) {
+        return builder().credentialsProviders(awsCredentialsProviders).build();
+    }
+
     @Override
     public AwsCredentials resolveCredentials() {
         if (reuseLastProviderEnabled && lastUsedProvider != null) {
-            return lastUsedProvider.resolveCredentials();
+            // TODO: Exception handling for join?
```

Comment:
What are the plans to handle this TODO? Will we have a separate PR? Also, I'm just wondering why we can't have a separate synchronous/blocking resolveIdentity API if we want to resolve the identity immediately. That API could internally handle the interruptions for join, instead of caller doing explicit joins and handling exceptions.

Reply:
Yes, handling joins and exception handling has been punted for future PR to address it holistically.